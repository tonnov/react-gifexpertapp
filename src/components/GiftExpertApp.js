import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState(['matrix']);

    // const handleAdd = () => {
    //     const rnd = Math.floor(Math.random() * 100);
    //     setCategories( cat => [...cat, `val ${rnd}`]);
    // }

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            {/* <hr /> */}
            <ol>
                { 
                    categories.map( category => (
                        <GifGrid key={category} category={category} />
                    ))
                }
            </ol>
        </>
    )
}
