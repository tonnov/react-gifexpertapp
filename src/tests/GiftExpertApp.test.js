import React from 'react'
import { shallow } from 'enzyme'
import { GiftExpertApp } from '../GiftExpertApp'

describe('Test suite for GifExpertApp', () => {
    
    test('should match snapshot', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should render a category list', () => {
        const categories = ['matrix', 'leon'];
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
