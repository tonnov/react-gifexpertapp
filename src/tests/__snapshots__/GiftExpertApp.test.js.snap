// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for GifExpertApp should match snapshot 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Test suite for GifExpertApp should render a category list 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="matrix"
      key="matrix"
    />
    <GifGrid
      category="leon"
      key="leon"
    />
  </ol>
</Fragment>
`;
